src/components/plots/rectanglePlot.ts

```diff
@@ -33,26 +42,39 @@ export module Plots {
       var attrToProjector = super._generateAttrToProjector();
 
       // Copy each of the different projectors.
-      var x1Attr = attrToProjector["x1"];
-      var y1Attr = attrToProjector["y1"];
-      var x2Attr = attrToProjector["x2"];
-      var y2Attr = attrToProjector["y2"];
+      var xAttr = attrToProjector[Rectangle._X_KEY];
+      var x1Attr = attrToProjector[Rectangle._X1_KEY];
+      var yAttr = attrToProjector[Rectangle._Y_KEY];
+      var y1Attr = attrToProjector[Rectangle._Y1_KEY];
 
-      // Generate width based on difference, then adjust for the correct x origin
-      attrToProjector["width"] = (d, i, dataset, m) => Math.abs(x2Attr(d, i, dataset, m) - x1Attr(d, i, dataset, m));
-      attrToProjector["x"] = (d, i, dataset, m) => Math.min(x1Attr(d, i, dataset, m), x2Attr(d, i, dataset, m));
+      var xScale = this.x().scale;
+      var yScale = this.y().scale;
 
-      // Generate height based on difference, then adjust for the correct y origin
-      attrToProjector["height"] = (d, i, dataset, m) => Math.abs(y2Attr(d, i, dataset, m) - y1Attr(d, i, dataset, m));
-      attrToProjector["y"] = (d, i, dataset, m) => {
-        return Math.max(y1Attr(d, i, dataset, m), y2Attr(d, i, dataset, m)) - attrToProjector["height"](d, i, dataset, m);
-      };
+      // In order to define a range from x, x1 has to also be set.
+      // If x1 is not set, auto-centering logic will be applied to x.
+      // The above also applies to y/y1.
```

Comment:
Let's fold the last line comment here into the first 2.

As a side note, I don't really know if this comment is necessary